### ISSUE TITLE
Integer division issue in python 3 for texttiling

### ISSUE THREAD

**Issue by arranger1044** [nltk/nltk#1209]
When computing the clip index under python 3, when calling the `tokenize` method of a `TextTilingTokenizer`, I receive the sequent error:

```
/usr/local/lib/python3.4/dist-packages/nltk/tokenize/texttiling.py in tokenize(self, text)
    125 
    126         # Boundary identification
--> 127         depth_scores = self._depth_scores(smooth_scores)
    128         segment_boundaries = self._identify_boundaries(depth_scores)
    129 

/usr/local/lib/python3.4/dist-packages/nltk/tokenize/texttiling.py in _depth_scores(self, scores)
    316         index = clip
    317 
--> 318         for gapscore in scores[clip:-clip]:
    319             lpeak = gapscore
    320             for score in scores[index::-1]:

TypeError: slice indices must be integers or None or have an __index__ method
```

From what I can see from the source code it may happen that the clip index is indeed a floating point and not an integer:

```
def _depth_scores(self, scores):
        """Calculates the depth of each gap, i.e. the average difference
        between the left and right peaks and the gap's score"""

        depth_scores = [0 for x in scores]
        #clip boundaries: this holds on the rule of thumb(my thumb)
        #that a section shouldn't be smaller than at least 2
        #pseudosentences for small texts and around 5 for larger ones.

        clip = min(max(len(scores)/10, 2), 5)
```

Can someone confirm this?

**Comment by rjwebb** [comment]
The result of division of two ints under Python 3 is always a float. So depending on the value of `len(scores)`, clip can be either a float or an int... This particular part of the source was written back in 2010 (https://github.com/nltk/nltk/commit/9f4243a633917b36689b2305779e377ef91ec29e), so I assume it was written with Python 2 in mind (where the result of integer division was rounded down).

**Comment by queirozfcom** [comment]
I was going to fix this but apparently it's already been fixed in branch `develop`

**Comment by alvations** [comment]
Yes this should have been resolved in #1863
